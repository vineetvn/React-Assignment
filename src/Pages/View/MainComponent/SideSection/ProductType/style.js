import styled from 'styled-components';
import {Link} from 'react-router-dom';

export const Div = styled.div `
    flex:25%auto;
    margin: 130px 50px 0 50px;
    display: flex;
    flex-direction: column;
    text-align: center;
    background-color: ${props => props.theme.color.silverGrey};
    color: ${props => props.theme.color.darkCharcoal};
    font-size: ${props => props.theme.fontSize.productName};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    padding: 30px 10px 10px 10px;

    @media only screen and (max-width: 1000px) {
        margin: 25px;
        flex: 25%;

    }
`

export const StyledLink = styled(Link) `
    text-decoration: none;
    color: ${props => props.theme.color.red};
    font-weight: ${props => props.theme.fontWeight.bold};
    padding-bottom: 20px;
`