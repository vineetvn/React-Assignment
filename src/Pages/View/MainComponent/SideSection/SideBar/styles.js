import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const Div = styled.div `
    flex: 25%;
    margin: 50px 10px;
    font-family: ${props => props.theme.fontFamily.Montserrat};

    @media only screen and (max-width: 500px) {
        margin: 30px;
    }
`

export const H3 = styled.h3 `
    font-family: ${props => props.theme.fontFamily.Montserrat};
`