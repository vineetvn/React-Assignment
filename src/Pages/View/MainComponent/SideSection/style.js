import styled from 'styled-components';

import Woman from '../../../../Assets/Images/Woman.jpg';

export const ImageWrapper = styled.div `
    background: url('${Woman}');
    flex: 75%;
    background-repeat: no-repeat;
    background-position: center;
    height: 100%auto;
    margin: 50px;

    @media only screen and (max-width: 1000px) {
        margin: 25px;
        flex: 75%;
        height: 100%;
    }    
`

export const Div = styled.div `
    display: flex;
    flex-direction: column;
    flex: 30%;

    @media only screen and (max-width: 1000px) {
        margin-top: 100px;
    } 
`