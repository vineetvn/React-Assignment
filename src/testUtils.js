import { mount, shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store'
import { ThemeProvider } from 'styled-components';
import {theme} from './Theme/Theme'

export const shallowRender = (
    Component,
    withRouter = false,
    props = {},
    dive = false
) => {
    if(!Component) return;

    let wrapperComponent = !withRouter ? shallow(<Component {...props} />)
    : shallow(<Component.WrappedComponent {...props} />);

    if(dive) {
        return wrapperComponent.dive();
    }else return wrapperComponent;
}


export const configureMockStore = (initialState = {}) => {  
    const mockStore = configureStore();
  
    const store = mockStore(initialState);
  
    return store;
};

export const mountRenderWithRouter = (MyComponent, store, props) => {
    return mount(
        <BrowserRouter>
            <ThemeProvider theme = {theme}>
                <Provider store = {store}>
                    <MyComponent {...props} />
                </Provider>
            </ThemeProvider>
        </BrowserRouter>
    );
  };
