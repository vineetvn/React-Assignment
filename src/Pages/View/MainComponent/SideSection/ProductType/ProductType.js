import React from 'react'
import { Div, StyledLink } from './style'
import {connect} from 'react-redux'

function ProductType(props) {
    let name;
    props.relatedProducts.filter(prod => {
        if(prod.id === 0) name = prod.name;
        return 0;
    })
    return (
        <Div>
            <h2>Welcome To The Shop Section</h2>
            {props.locationState + 1 ? <h3>{name} Section</h3> : null}
            <p>Return back to  <StyledLink to='/home'>Home</StyledLink></p>
        </Div>
    )
}

const mapStateToProps = state => {
    return {
        relatedProducts: state.view.relatedProducts,
        locationState: state.view.locationState
    }
}

export default connect(mapStateToProps, null)(ProductType)
