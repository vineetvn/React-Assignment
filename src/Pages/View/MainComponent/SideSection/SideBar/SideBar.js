import React, { Component } from 'react'
import { Div, H3 } from './styles'
import { connect } from 'react-redux'
import Card from './Card/Card';

export class SideBar extends Component {

    render() {
        let products = this.props.relatedProducts.filter(prod => {
            if(prod.id < 4) return true;
            else return false;
        })

        let popular = products.map(prod => {
            return <Card key={prod.id} img={prod.url} name={prod.name} price={prod.price} />
        })

        return (
            <Div>
                <H3>Popular Products</H3>
                {popular}
            </Div>
        )
    }
}

const mapStateToProps = state => {
    return {
        relatedProducts: state.view.relatedProducts,
        productId: state.prod.clicked,
        color: state.color.color
    }
}

export default connect(mapStateToProps, null)(SideBar)
