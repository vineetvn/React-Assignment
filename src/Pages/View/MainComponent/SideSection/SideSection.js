import React from 'react'
import { connect } from 'react-redux';

import ProductType from './ProductType/ProductType';
import SideBar from './SideBar/SideBar';
import {ImageWrapper, Div} from './style';

function SideSection(props) {
    return (
        props.locationState + 1 ? <Div>
            <ProductType />
            <ImageWrapper />
        </Div> : <SideBar />
    )
}

const mapStateToProps = state => {
    return {
        locationState: state.view.locationState
    }
}

export default connect(mapStateToProps, null)(SideSection)
