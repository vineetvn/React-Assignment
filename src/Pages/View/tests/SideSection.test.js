import React from 'react'

import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme';
import { configureMockStore, mountRenderWithRouter, shallowRender } from '../../../testUtils';
import {SideSection as SS} from '../MainComponent/SideSection/SideSection'
import SideSection  from '../MainComponent/SideSection/SideSection'
import { SideBar } from '../MainComponent/SideSection/SideBar/SideBar';
import ProductType from '../MainComponent/SideSection/ProductType/ProductType';



configure({adapter: new Adapter()});

describe('SideView', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowRender(SS);
    })

    it('should render the component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should render a sideBar with products when navigated to the shop directly', () => {
        expect(wrapper.find(SideBar));
    })

    it('should render a sidebar with image when clicked on the products', () => {
        wrapper.setProps({locationState: true})
        expect(wrapper.find(ProductType));
    })
})

describe('SideSection deep render', () => {
    let initialState = {
        view: {
            relatedProducts: [
                {
                    id: 0,
                    url: 'bagpack.jpg',
                    name: 'backpack',
                    price: '$10'
                }
            ],
            locationState: 0
        }
    }

    let props = {
        relatedProducts: [
            {
                id: 0,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            }
        ],
        locationState: 0,
        img: 'Backpack.jpg',
        name: 'BackPack',
        price: '$10'
    }

    let store = configureMockStore(initialState)
    let wrapper;

    beforeEach(() => {
        wrapper = mountRenderWithRouter(SideSection, store, props)
    })

    it('should render deeply', () => {
        expect(wrapper).toMatchSnapshot();
    });
});