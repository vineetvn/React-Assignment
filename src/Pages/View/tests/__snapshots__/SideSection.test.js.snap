// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideSection deep render should render deeply 1`] = `ReactWrapper {}`;

exports[`SideView should render the component 1`] = `ShallowWrapper {}`;
